src/storage/src/source/kafka.rs

```diff
@@ -948,6 +949,32 @@ fn construct_source_message(
                     .into();
                 packer.push(d)
             }
+            KafkaMetadataKind::Header { key, use_bytes } => {
+                match msg.headers() {
+                    Some(headers) => {
+                        let d = headers
+                            .iter()
```

Comment:
In the thread of #20958 we discussed defining the semantics of what happens when multiple headers with the same key are present (e.g first/last wins). The implementation here surfaces the first value that happens to be in the message which is a bit more vague. Should we sort the headers before picking the first that matches?

Reply:
> The implementation here surfaces the first value that happens to be in the message which is a bit more vague. Should we sort the headers before picking the first that matches?

Er, sort how? IMO, Kafka message headers are explicitly an ordered list, and so that's the order we should respect! Of course, still need to define whether we want first wins or last wins semantics. IMO, last wins semantics is a bit more standard (e.g., that's how duplicated CLI args usually work).

Reply:
> Kafka message headers are explicitly an ordered list,

That is their encoding but that's not necessarily their semantic meaning. If kafka headers have the semantics of `HashMap<bytes, HashSet<bytes>>` then we ought to sort possible values of a key and pick the first one so when the same set gets inserted the same value gets picked. If the kafka headers have the semantics of `HashMap<bytes, Vec<bytes>>` then the current implementation is correct. Are you asserting that Kafka has the latter semantic meaning or inferring it from the encoding?

Reply:
It _is_ their semantic meaning. See [KIP-82](https://cwiki.apache.org/confluence/display/KAFKA/KIP-82+-+Add+Record+Headers#KIP82AddRecordHeaders-Keyduplicationandordering) and the [`Headers`](https://kafka.apache.org/24/javadoc/index.html?org/apache/kafka/common/header/Headers.html) Java API. The API explicitly preserves ordering. That's why our representation of the headers column created by `INCLUDE HEADERS` is `(key: bytea, value: bytea) list`.

I just found more evidence for why `INCLUDE HEADER` use last-wins semantics in the order in which `Headers.add` was called: because there is a [`Headers.lastHeader`](https://kafka.apache.org/24/javadoc/index.html?org/apache/kafka/common/header/Headers.html) method but no `Headers.firstHeader` method.

Reply:
That's a great callout. The [implementation in the Java client](https://github.com/apache/kafka/blob/1e7e9ce91897e03f5611cd74f6e34f723a586ab6/clients/src/main/java/org/apache/kafka/common/header/internals/RecordHeaders.java#L85) indeed uses a list to preserve ordering as suggested by the KIP. I'll change the semantics to last wins.